tests/foreman/cli/test_errata.py

```diff
@@ -714,757 +700,768 @@ def test_host_errata_search_commands(self):
-        """Filter errata by product id and Org id
-
-        :id: caf14671-d8b2-4a23-8c7e-6667bb78d4b7
-
-        :Setup: Errata synced on satellite server.
-
-        :Steps: erratum list --product-id=<product_id>
-            --organization-id=<org_id>
-
-        :expectedresults: Errata is filtered by product id and Org id.
-        """
-        product_erratum_list = Erratum.list(
+    )
+    Repository.synchronize({'id': repo['id']})
+    return org1
+
+
+@pytest.fixture(scope='class')
+def org2():
+    """an org with one custom product & repository"""
+    org2 = entities.Organization().create()
+    org2.product = entities.Product(organization=org2).create()
+    repo = make_repository(
+        {
+            'download-policy': 'immediate',
+            'organization-id': org2.id,
+            'product-id': org2.product.id,
+            'url': FAKE_1_YUM_REPO,
+        }
+    )
+    Repository.synchronize({'id': repo['id']})
+    return org2
+
+
+@pytest.fixture(scope='class')
+def org3():
+    """an org with two custom products with repository"""
+    # product small
+    org3 = entities.Organization().create()
+    org3.product_small = make_product(options={'organization-id': org3.id})
+    repo = make_repository(
+        {
+            'download-policy': 'immediate',
+            'organization-id': org3.id,
+            'product-id': org3.product_small['id'],
+            'url': FAKE_2_YUM_REPO,
+        }
+    )
+    Repository.synchronize({'id': repo['id']})
+    # product big
+    org3.product_big = make_product(options={'organization-id': org3.id})
+    repo = make_repository(
+        {
+            'download-policy': 'immediate',
+            'organization-id': org3.id,
+            'product-id': org3.product_big['id'],
+            'url': FAKE_3_YUM_REPO,
+        }
+    )
+    Repository.synchronize({'id': repo['id']})
+    return org3
+
+
+def _get_sorted_erratum_ids_info(erratum_ids, sort_by='issued', sort_reversed=False):
+    """Query hammer for erratum ids info
+
+    :param erratum_ids: a list of errata id
+    :param sort_by: the field to sort by the results (issued or updated)
+    :param sort_reversed: whether the sort should be reversed
+            (not ascending)
+    :return: a list of errata info dict for each errata id in erratum_ids
+    :type erratum_ids: list[str]
+    :type sort_by: str
+    :type sort_reversed: bool
+    :rtype: list[dict]
+    """
+    if len(erratum_ids) > ERRATUM_MAX_IDS_INFO:
+        raise Exception('Erratum ids length exceeded')
+    erratum_info_list = []
+    for errata_id in erratum_ids:
+        erratum_info_list.append(Erratum.info(options={'id': errata_id}, output_format='json'))
+    sorted_erratum_info_list = sorted(
+        erratum_info_list, key=itemgetter(sort_by), reverse=sort_reversed
+    )
+    return sorted_erratum_info_list
+
+
+@pytest.mark.tier3
+@pytest.mark.upgrade
+def test_positive_list_sort_by_issued_date(module_org):
```

Comment:
These test functions are defined after `TestErrata` class, but aren't indented under it. Earlier comments about including `self` apply here as well if these test cases should be inside the class.

Reply:
Hi

I have removed the classes
I have reworked the fixtures, please review again